SimTKmath/Geometry/src/ContactGeometryImpl.h

```diff
@@ -132,13 +137,34 @@ class SimTK_SIMMATH_EXPORT ContactGeometryImpl {
                         Vec3& closestPointOnLine,
                         Real& height) const;
 
+    bool calcNearestPointOnLineImplicitly(
+            const Vec3& pointA,
+            const Vec3& pointB,
+            Vec3& nearestPointOnLine,
+            size_t maxIter,
+            Real tolerance) const;
+
     // Geodesic evaluators
 
 
     // Given two points, find a geodesic curve connecting them.
     void initGeodesic(const Vec3& xP, const Vec3& xQ, const Vec3& xSP,
             const GeodesicOptions& options, Geodesic& geod) const;
 
+    void shootGeodesicInDirectionImplicitly(
+            Vec3 pointApprox,
+            Vec3 tangentApprox,
+            Real finalArcLength,
+            Real& initStepSize,
+            Real integratorAccuracy,
+            Real constraintTolerance,
+            int maxIter,
+            Vec2& finalJacobi,
+            Vec2& finalJacobiDot,
+            std::function<void(
```

Comment:
I've added a struct `ContactGeometry::GeodesicKnotPoint` for this. Still not exactly amazing, because it could be better integrated with the state used by the `GeodesicIntegrator`, (again, see https://github.com/simbody/simbody/pull/791#discussion_r1663898202)